src/public/map_factory.ts

```diff
@@ -87,7 +91,10 @@ export const map = (domElement: HTMLElement | string, apiKey: string, options?:
   const mapId = _mapObjects.length;
   const mapApiObject = new EmscriptenApi(wrldModule);
   const mapOptions = options || {};
-  const onMapRemove = () => { delete _mapObjects[mapId]; };
+  const onMapRemove = () => {
+    delete _mapObjects[mapId];
+    abortInitializingMap(wrldModule);
```

Comment:
Do we need to hold a copy of all the wrld modules as mapObjects is an array?

Reply:
It's used by `getMapById`. I didn't add that, just moved it to its own line now the function's more complicated.